llvm/lib/Transforms/Utils/CodeExtractor.cpp

```diff
@@ -627,6 +627,26 @@ bool CodeExtractor::isEligible() const {
         return false;
     }
   }
+  // stacksave as input implies stackrestore in the outlined function.
+  // This can confuse prolog epilog insertion phase.
+  // stacksave's uses must not cross outlined function.
+  for (BasicBlock *BB : Blocks) {
+    for (Instruction &I : *BB) {
+      IntrinsicInst *II = dyn_cast<IntrinsicInst>(&I);
+      if (!II)
+        continue;
+      bool IsSave = II->getIntrinsicID() == Intrinsic::stacksave;
+      bool IsRestore = II->getIntrinsicID() == Intrinsic::stackrestore;
+      if (IsSave && any_of(II->users(), [&Blks = this->Blocks](User *U) {
+            return !definedInRegion(Blks, U);
+          })) {
+        return false;
+      }
+      if (IsRestore && !definedInRegion(Blocks, II->getArgOperand(0))) {
+        return false;
+      }
```

Comment:
llvm coding style nit: single statement loops and if/else/etc shouldn't have `{}`s

Reply:
okay to keep the one around the body of the outer loop